### PR TITLE
abiword: 3.0.1 -> 3.0.2

### PR DESCRIPTION
**Pull request by ylwghst**
###### Motivation for this change

Two bugs both in 3.0.1 and 3.0.2 makes Abiword unusable.
Raising to 3.0.2 and appending fix patch.

**Bug 13791**
Bug description: constant redraw flicker when document not empty
Bug report: https://bugzilla.abisource.com/show_bug.cgi?id=13791
Patch source: https://bugzilla.abisource.com/attachment.cgi?id=5860

**Bug 13815**
Bug description: Document view is black
Bug report: https://bugzilla.abisource.com/show_bug.cgi?id=13815
Patch source: https://bugzilla.abisource.com/attachment.cgi?id=5837

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [+] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [+] Tested execution of all binary files (usually in `./result/bin/`)
- [+] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by ylwghst** [comment]
Changed patch to patch which combines both fixes.

File: pkgs/applications/office/abiword/default.nix [hunk 1]
```diff
@@ -5,15 +5,19 @@
...
-    url = "http://www.abisource.org/downloads/abiword/${version}/source/${name}.tar.gz";
-    sha256 = "1ik591rx15nn3n1297cwykl8wvrlgj78i528id9wbidgy3xzd570";
+    url = "http://www.abisource.com/downloads/abiword/${version}/source/${name}.tar.gz";
+    sha256 = "08imry821g81apdwym3gcs4nss0l9j5blqk31j5rv602zmcd9gxg";
   };
 
   enableParallelBuilding = true;
 
+  patches = [
+    (fetchurl { url = https://gist.github.com/ylwghst/b28ef051980f41aca56df7a8affd9526/raw/; sha256 = "02p8kz02xm1197zcpzjs010mna9hxsbq5lwgxr8b7qhh9yxja7al"; })
```
Comment (grahamc) [inline pkgs/applications/office/abiword/default.nix]: I think we'd prefer this patch be in nixpkgs, than just in your gists.

Comment (grahamc) [reply pkgs/applications/office/abiword/default.nix]: In fact, just use two patches, linking to the original URLs you already put in the body of the issue?

**Comment by jtojnar** [comment]
```
    (fetchurl {
      url = https://bugzilla.abisource.com/attachment.cgi?id=5860;
      sha256 = "02p8kz02xm1197zcpzjs010mna9hxsbq5lwgxr8b7qhh9yxja7al";
    })
```
works just fine.

It looks like https://bugzilla.abisource.com/attachment.cgi?id=5837 is already merged.

**Comment by ylwghst** [comment]
@jtojnar I've put in code only nr. **5860** which fixes both issues.

**Comment by jtojnar** [comment]
@ylwghst Yes, it looks like that.

File: pkgs/applications/office/abiword/default.nix (hunk 1 above)
Comment (ylwghst) [reply pkgs/applications/office/abiword/default.nix]: When I try to fetch this URL: https://bugzilla.abisource.com/attachment.cgi?id=5860&format=raw
Patch is fetched as 'attachment.cgi?id=5860&format=raw' which is problem due to the & in it's name.

Comment (jtojnar) [reply pkgs/applications/office/abiword/default.nix]: You can add `name` attribute but it should not be necessary to add `format`.

**Comment by jtojnar** [comment]
We should still use the remote patch.

**Comment by Mic92** [comment]
also ported to stable.

**Pull request merged**